context/context.go

```diff
@@ -0,0 +1,52 @@
+package context
+
+import (
+	"golang.org/x/net/context"
+)
+
+// Context is a copy of Context from the golang.org/x/net/context package.
+type Context interface {
+	context.Context
+}
+
+// Background returns a non-nil, empty Context.
+func Background() Context {
+	return context.Background()
+}
+
+// WithValue returns a copy of parent in which the value associated with key is
+// val. Use context Values only for request-scoped data that transits processes
+// and APIs, not for passing optional parameters to functions.
+func WithValue(parent Context, key, val interface{}) Context {
+	return context.WithValue(parent, key, val)
+}
+
+// stringMapContext is a simple context implementation that checks a map for a
+// key, falling back to a parent if not present.
+type stringMapContext struct {
+	context.Context
+	m map[string]string
+}
+
+// WithMapContext returns a context that proxies lookups through a map.
+func WithMapContext(ctx context.Context, m map[string]string) context.Context {
```

Comment:
This stutters a bit. Should this just be called `WithValues`? Also, taking a `map[string]interface{}` will probably be okay. It's okay to change the type of `config.Log.Fields` to comply.